zimbra/src/main/java/fr/openent/zimbra/service/impl/MessageService.java

```diff
@@ -1011,6 +1073,64 @@ public void deleteMessages(List<String> listMessageIds, UserInfos user, boolean
         }
     }
 
+    /**
+     * Delete list of message Ids from trash
+     *
+     * @param messageIDs        List of Message ID
+     * @param user              User infos
+     * @param trashConstraint   boolean constraint that the message should be in trash
+     */
+    public Future<Void> deleteMessages(List<String> messageIDs, UserInfos user, boolean trashConstraint) {
+        Promise<Void> promise = Promise.promise();
+        List<Future<Void>> futures = new ArrayList<>();
+
+        List<List<String>> batchMessageIds = ArrayHelper.split(messageIDs, BATCH_SIZE);
+        for (List<String> ids : batchMessageIds) {
+            futures.add(this.deleteBatchMessages(ids, user, trashConstraint));
+        }
```

Comment:
List<List<String>> batchMessageIds = ArrayHelper.split(messageIDs, BATCH_SIZE).stream()
.map(ids -> this.deleteBatchMessages(ids, user, trashContraint))